### DISCUSSION TITLE
When using the Libero-TCL flow to program the PF SOC, I don't see the HSS eNVM Client

### DISCUSSION THREAD

**Discussion by haroldnelson-2021** [polarfire-soc/polarfire-soc-documentation#458] [HSS and U-Boot] [answered]
I periodically program my Icicle board using the Libero-TCL flow, but recently reviewed the memory clients and do not see an HSS image being loaded into eNVM.  

1. Am I just fortunate to never have seen an issue? Or, if using Bootmode0 the HSS gets loaded into Scratchpad-LIM by SoftConsole?
2. However, when SoftConsole is not being used, the board still boots Linux fine, etc.  In this case is HSS part of the SD card's image and there is only a simple bootloader in the PF SOC?

**Comment by hughbreslin** [comment]
Hi @haroldnelson-2021 The tools won't overwrite an existing HSS programmed into the device or change the boot mode unless you set it to. If you had originally programmed the HSS via a reference design or custom build it will stay in eNVM until overwritten. If you change from boot mode 1 to boot mode 0 you're only changing the boot set up and not the eNVM content.

To get the latest HSS you will either need to update using a job file provided with the reference design or build the HSS manually. Otherwise your HSS version will stay fixed at the last one used in an update while your design gets updated to newer versions.

**Reply by haroldnelson-2021** [reply]
Hugh, thanks.  Is there a reason for not including the eNVM memory clients in the TCL build, just wondering why that would not be the default?

**Reply by hughbreslin** [reply, marked as answer]
The memory client is an output of the HSS build which is separate. It could be included using the bitstream builder or the Tcl argument where you specify the location of the hex file but theres no guarantee softconsole would be installed when we run the Tcl script and we can't store the HSS hex file in the git repo as its not a source
